multimodal/src/autogluon/multimodal/models/utils.py

```diff
@@ -686,28 +686,45 @@ def update_mmdet_config(key, value, config):
                     update_mmdet_config(key, value, subsubconfig)
 
 
-def run_model(model: nn.Module, batch: dict):
+def run_model(model: nn.Module, batch: dict, trt_model: Optional[nn.Module] = None):
+    from ..utils.onnx import OnnxModule
     from .document_transformer import DocumentTransformer
+    from .fusion.fusion_mlp import MultimodalFusionMLP
     from .huggingface_text import HFAutoModelForTextPrediction
+    from .t_few import TFewModel
     from .timm_image import TimmAutoModelForImagePrediction
 
-    supported_models = (TimmAutoModelForImagePrediction, HFAutoModelForTextPrediction)
+    supported_models = (
+        TimmAutoModelForImagePrediction,
+        HFAutoModelForTextPrediction,
+        MultimodalFusionMLP,
+        TFewModel,
+        OnnxModule,
+    )
     pure_model = model.module if isinstance(model, nn.DataParallel) else model
+    if isinstance(pure_model, OnnxModule):
+        for k in batch:
```

Comment:
What possible types can `batch[k].dtype` be when code runs here? Can we say it is always in `[torch.float32, torch.int32, torch.int64]`?

Reply:
This is a special handling for converting the data type in token_ids. I'm not sure why the token_ids are in int32, but the inputs are required to be int64.